src/main/java/org/sagebionetworks/web/client/widget/header/HeaderViewImpl.java

```diff
@@ -102,22 +102,23 @@ public void setPresenter(Presenter presenter) {
 		refresh();
 	}
 
-	private void detachNavBar() {
```

Comment:
Since the ReactComponentDiv now more properly mounts/unmounts the component, attaching and detaching caused undesired flashing. Let's just show/hide it and pass new props instead of completely detaching and re-attaching.